File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/AbstractPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace gov.va.medora.mdo.domain.sm
{
    public abstract class AbstractPage<T> : IPage<T>
    {
        protected int _pageSize;
        protected int _pageNumber;
        protected List<T> _elements;
        protected int _elementCount;

        public int ElementCount
        {
            get { return _elementCount; }
            set { _elementCount = value; }
        }
        public List<T> Elements
        {
            get { return _elements; }
            set { _elements = value; }
        }
        public int PageNumber
        {
            get { return _pageNumber; }
            set { _pageNumber = value; }
        }
        public int PageSize
        {
            get { return _pageSize; }
            set { _pageSize = value; }
        }

        public int getLastPageNumber()
        {
            if (_elementCount == 0) 
            { 
                return 0; 
            }
            return Convert.ToInt32(Math.Floor((double)((_elementCount - 1) / _pageSize)));
        }

        public int getNextPageNumber()
        {
            return _pageNumber + 1;
        }

        public int getPreviousPageNumber()
        {
            return _pageNumber - 1;
        }

        public int getThisPageFirstElementNumber()
        {
            return _pageNumber * _pageSize + 1;
        }

        public int getThisPageLastElementNumber()
        {
            int fullPage = getThisPageFirstElementNumber() + _pageSize - 1;
            return _elementCount < fullPage ? _elementCount : fullPage;
        }

        public bool hasNextPage()
        {
            return !isLastPage();
        }

        public bool hasPreviousPage()
        {
            return _pageNumber > 0;
        }

        public bool isFirstPage()
        {
            return _pageNumber == 0;
        }

        public bool isLastPage()
        {
            return _pageNumber >= getLastPageNumber();
        }



        public List<T> getElements()
        {
            throw new NotImplementedException();
        }

        public int getElementCount()
        {
            throw new NotImplementedException();
        }

        public void setElementCount(int total)
        {
            throw new NotImplementedException();
        }

        public void setElements(List<T> elements)
        {
            throw new NotImplementedException();
        }

        public int getPageSize()
        {
            throw new NotImplementedException();
        }

        public void setPageSize(int pageSize)
        {
            throw new NotImplementedException();
        }

        public int getPageNumber()
        {
            throw new NotImplementedException();
        }

        public void setPageNumber(int pageNumber)
        {
            throw new NotImplementedException();
        }
    }
}

File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/trash/AbstractPage.java
package gov.va.med.mhv.sm.model;

import java.util.List;

public abstract class AbstractPage<T> implements Page<T> {

	protected int pageSize;
	protected int pageNumber;
	protected List<T> elements;
	protected int elementCount;

	public int getElementCount() {
		return elementCount;
	}
	public void setElementCount(int total) {
		this.elementCount = total;
	}
	public List<T> getElements() {
		return elements;
	}
	public void setElements(List<T> elements) {
		this.elements = elements;
	}
	public int getPageNumber() {
		return pageNumber;
	}
	public void setPageNumber(int pageNumber) {
		this.pageNumber = pageNumber;
	}
	public int getPageSize() {
		return pageSize;
	}
	public void setPageSize(int pageSize) {
		this.pageSize = pageSize;
	}

	public int getLastPageNumber()
	{
		double totalResults = new Integer(elementCount).doubleValue();
		if(totalResults==0){return 0;}
		return new Double(Math.floor((totalResults-1) / pageSize)).intValue();
	}

	public int getNextPageNumber() {
		return pageNumber + 1;
	}

	public int getPreviousPageNumber() {
		return pageNumber - 1;
	}

	public int getThisPageFirstElementNumber() {
		return pageNumber * pageSize + 1;
	}

	public int getThisPageLastElementNumber() {
		int fullPage = getThisPageFirstElementNumber() + pageSize - 1;
	    return elementCount < fullPage ? elementCount : fullPage;
	}

	public boolean hasNextPage() {
		return !isLastPage();
	}

	public boolean hasPreviousPage() {
		return pageNumber > 0;
	}

	public boolean isFirstPage() {
		return pageNumber == 0;
	}

	public boolean isLastPage() {
		return pageNumber >= getLastPageNumber();
	}









}
